### ISSUE TITLE
AVVideoSettings crashes with CMVideoCodecType enum error

### ISSUE THREAD

**Issue by tomfisher** [robovm/robovm#1173]
Hi, I am trying to use AVAssetWriterInput to generate a movie from a couple of pictures with the following code:

```
		AVVideoSettings videoSettings = new AVVideoSettings();
		videoSettings.setCodec(CMVideoCodecType.H264);
		videoSettings.setWidth(params.getWidth());
		videoSettings.setHeight(params.getHeight());
		AVAssetWriterInput videoInput = new AVAssetWriterInput(AVMediaType.Video, videoSettings);
``

But got an exception:

```
2015-10-18 18:27:19.076 NativeDemo[63080:719190] -[__NSCFNumber length]: unrecognized selector sent to instance 0xb000000617663313
2015-10-18 18:27:19.159 NativeDemo[63080:719190] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFNumber length]: unrecognized selector sent to instance 0xb000000617663313'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000108fccf65 __exceptionPreprocess + 165
	1   libobjc.A.dylib                     0x000000010887bdeb objc_exception_throw + 48
	2   CoreFoundation                      0x0000000108fd558d -[NSObject(NSObject) doesNotRecognizeSelector:] + 205
	3   CoreFoundation                      0x0000000108f22f7a ___forwarding___ + 970
	4   CoreFoundation                      0x0000000108f22b28 _CF_forwarding_prep_0 + 120
	5   AVFoundation                        0x00000001059bbc56 AVOSTypeForString + 39
	6   AVFoundation                        0x00000001059e84ac -[AVAVVideoSettingsVideoOutputSettings initWithAVVideoSettingsDictionary:exceptionReason:] + 863
	7   AVFoundation                        0x00000001059e7bf3 +[AVAVVideoSettingsVideoOutputSettings _videoOutputSettingsWithVideoSettingsDictionary:exceptionReason:] + 62
	8   AVFoundation                        0x00000001059e1e84 +[AVOutputSettings _outputSettingsWithOutputSettingsDictionary:exceptionReason:] + 441
	9   AVFoundation                        0x00000001059e6adf +[AVVideoOutputSettings _videoOutputSettingsWithVideoSettingsDictionary:exceptionReason:] + 44
	10  AVFoundation                        0x00000001059e1e84 +[AVOutputSettings _outputSettingsWithOutputSettingsDictionary:exceptionReason:] + 441
	11  AVFoundation                        0x0000000105992446 -[AVAssetWriterInput initWithMediaType:outputSettings:sourceFormatHint:] + 383
	12  NativeDemo                          0x000000010187790b [J]org.robovm.apple.avfoundation.AVAssetWriterInput.$m$initWithMediaType$outputSettings$(Lorg/robovm/apple/avfoundation/AVAssetWriterInput;Lorg/robovm/objc/Selector;Lorg/robovm/apple/avfoundation/AVMediaType;Lorg/robovm/apple/foundation/NSDictionary;)J + 199
	13  NativeDemo                          0x000000010187542d [J]org.robovm.apple.avfoundation.AVAssetWriterInput.init(Lorg/robovm/apple/avfoundation/AVMediaType;Lorg/robovm/apple/foundation/NSDictionary;)J + 98
	14  NativeDemo                          0x000000010187497d [J]org.robovm.apple.avfoundation.AVAssetWriterInput.<init>(Lorg/robovm/apple/avfoundation/AVMediaType;Lorg/robovm/apple/avfoundation/AVVideoSettings;)V + 95
```

**Comment by tomfisher** [comment]
Have you confirmed whether this is a bug?? 

Just paste the following code to anywhere in your project:

```
    AVVideoSettings videoSettings = new AVVideoSettings();
    videoSettings.setCodec(CMVideoCodecType.H264);
    videoSettings.setWidth(1024);
    videoSettings.setHeight(1024);
    AVAssetWriterInput videoInput = new AVAssetWriterInput(AVMediaType.Video, videoSettings);
```

then you got something like: 

```
2015-11-30 15:20:14.764 NativeDemo[74338:415720] -[__NSCFNumber length]: unrecognized selector sent to instance 0xb000000617663313
2015-11-30 15:20:14.821 NativeDemo[74338:415720] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFNumber length]: unrecognized selector sent to instance 0xb000000617663313'
*** First throw call stack:
(
    0   CoreFoundation                      0x0000000116a75f45 __exceptionPreprocess + 165
    1   libobjc.A.dylib                     0x0000000116324deb objc_exception_throw + 48
    2   CoreFoundation                      0x0000000116a7e56d -[NSObject(NSObject) doesNotRecognizeSelector:] + 205
    3   CoreFoundation                      0x00000001169cbeea ___forwarding___ + 970
    4   CoreFoundation                      0x00000001169cba98 _CF_forwarding_prep_0 + 120
    5   AVFoundation                        0x00000001133ce580 AVOSTypeForString + 39
    6   AVFoundation                        0x00000001133faf04 -[AVAVVideoSettingsVideoOutputSettings initWithAVVideoSettingsDictionary:exceptionReason:] + 863
    7   AVFoundation                        0x00000001133fa64b +[AVAVVideoSettingsVideoOutputSettings _videoOutputSettingsWithVideoSettingsDictionary:exceptionReason:] + 62
    8   AVFoundation                        0x00000001133f48df +[AVOutputSettings _outputSettingsWithOutputSettingsDictionary:exceptionReason:] + 441
    9   AVFoundation                        0x00000001133f9537 +[AVVideoOutputSettings _videoOutputSettingsWithVideoSettingsDictionary:exceptionReason:] + 44
    10  AVFoundation                        0x00000001133f48df +[AVOutputSettings _outputSettingsWithOutputSettingsDictionary:exceptionReason:] + 441
    11  AVFoundation                        0x00000001133a4d2c -[AVAssetWriterInput initWithMediaType:outputSettings:sourceFormatHint:] + 383
    12  NativeDemo                          0x000000010ee1526b [J]org.robovm.apple.avfoundation.AVAssetWriterInput.$m$initWithMediaType$outputSettings$(Lorg/robovm/apple/avfoundation/AVAssetWriterInput;Lorg/robovm/objc/Selector;Lorg/robovm/apple/avfoundation/AVMediaType;Lorg/robovm/apple/foundation/NSDictionary;)J + 199
    13  NativeDemo                          0x000000010ee12dad [J]org.robovm.apple.avfoundation.AVAssetWriterInput.init(Lorg/robovm/apple/avfoundation/AVMediaType;Lorg/robovm/apple/foundation/NSDictionary;)J + 98
    14  NativeDemo                          0x000000010ee12300 [J]org.robovm.apple.avfoundation.AVAssetWriterInput.<init>(Lorg/robovm/apple/avfoundation/AVMediaType;Lorg/robovm/apple/avfoundation/AVVideoSettings;)V + 95
```

**Comment by tomfisher** [comment]
This is resolved in RoboVm 1.12
